apps/beeai-ui/src/components/LineClampText/LineClampText.tsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright 2025 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ExpandButton } from '#components/ExpandButton/ExpandButton.tsx';
+import clsx from 'clsx';
+import { CSSProperties, PropsWithChildren, useCallback, useEffect, useId, useRef, useState } from 'react';
+import { useDebounceCallback } from 'usehooks-ts';
+import classes from './LineClampText.module.scss';
+
+interface Props {
+  lines: number;
+  className?: string;
+}
+
+export function LineClampText({ lines, className, children }: PropsWithChildren<Props>) {
+  const id = useId();
+  const textRef = useRef<HTMLElement>(null);
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [showButton, setShowButton] = useState(false);
+
+  const checkOverflow = useCallback(() => {
+    const element = textRef.current;
+
+    if (!element) {
+      return;
+    }
+
+    const lineHeight = parseFloat(getComputedStyle(element).lineHeight);
+    const height = lineHeight * lines;
+    const { scrollHeight } = element;
+
+    if (scrollHeight === 0) {
```

Comment:
Looks like a Guard clause that could be moved to the top, before parsing, and avoid unnecessary parsing.